### PR TITLE
Update Cosmos SDK - First pass

### PR DESCRIPTION
**Pull request by ToasterTheBrave**
* Update the version of lens to be a snapshot tag for now.  This will need changed again after all testing is complete
* Change to cosmos-sdk v0.47.1
* Change out all usages of github.com/gogo/protobuf to newer github.com/cosmos/gogoproto
* Small tweaks to support the new versions

# Related Github tickets

- a list of github issues that are relevant for this PR

# Background

_A short paragraph about what exactly this PR solves and what is the problem._

# Testing completed

- [ ] _a list of actions that you've performed to ensure that this PR works as intended_
- [ ] _99% of the times you should have an item: "wrote tests"_

### CODE REVIEW

**Comment by ToasterTheBrave** [comment]
@0xmuralik Would you mind giving this a review and testing this?  I am also in the first testing phase.

**Comment by taariq** [comment]
Also adding @aleem1314 to this review as well.

**Comment by ToasterTheBrave** [comment]
This is very much Draft.  There will be more that needs done, but a review and testing is still needed at this point.

**Comment by ToasterTheBrave** [comment]
https://github.com/palomachain/paloma/issues/795

**Review by taariq — APPROVED** [review]
LFG! Sparta!

File: chain/client.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func NewChainClient(ccc *lens.ChainClientConfig, input io.Reader, output io.Writ
 		Config:         ccc,
 		Input:          input,
 		Output:         output,
-		Codec:          lens.MakeCodec(ccc.Modules),
+		Codec:          lens.MakeCodec(ccc.Modules, []string{}),
```
Comment (0xmuralik) [inline chain/client.go:47]: Do we need to add "ethermint" as an extra codec?
Before lens upgrade https://github.com/VolumeFi/lens-1/blob/v0.5.1/client/encoding.go
The `MakeCodec` func registers the interfaces to the ethermint codec by default.
But in the upgrade to sdk 0.47 https://github.com/VolumeFi/lens-1/blob/def8a6f4cbfdc30e39dd70b6f364422ef1f7c262/client/encoding.go#L31
We are required to add ethermint as an extra codec.
@ToasterTheBrave @taariq @aleem1314

**Pull request merged**